CA mesh CA expiration to it's own section (#16747)
hashicorp/consul: website/content/docs/agent/telemetry.mdx

**What to look for:** For a healthy cluster, you're looking for a `lastContact` lower than 200ms, `leader` > 0 and `candidate` == 0. Deviations from this might indicate flapping leadership.

> ### Certificate Authority Expiration

Reviewer: ```suggestion
### Certificate authority expiration
```
Headings should be sentence case per recent style guide changes.

| Metric Name                            | Description                                                             | Unit    | Type  |
| :------------------------- | :---------------------------------------------------------------------------------- | :------ | :---- |

> | `consul.mesh.active-root-ca.expiry`    | The number of seconds until the root CA expires, updated every hour.    | seconds | gauge |

Reviewer: ```suggestion
| `consul.mesh.active-root-ca.expiry` | The number of seconds until the root CA expires. Consul updates the value every hour. | seconds | gauge |
```
Active voice

> | `consul.mesh.active-signing-ca.expiry` | The number of seconds until the signing CA expires, updated every hour. | seconds | gauge |

Reviewer: ```suggestion
| `consul.mesh.active-signing-ca.expiry` | The number of seconds until the signing CA expires. Consul updates the value every hour. | seconds | gauge |
```

> | `consul.mesh.active-signing-ca.expiry` | The number of seconds until the signing CA expires. Consul updates the value every hour. | seconds | gauge |

Reviewer 2: Optional: Consider linking to how to find out what the intermediate TTL configuration is for the CA system

Author: Maybe to this doc?  https://developer.hashicorp.com/consul/docs/connect/ca/consul#intermediatecertttl

> ** Why they're important:** An externally managed CA, like with Vault, means that the renewal is handled outside of Consul and it will need to be monitored for expiration (and rotated as appropriate). 

Reviewer: ```suggestion
** Why they're important:** When using an external system, such as Vault, to manage the CA, you should monitor renewals to ensure that certificates do not expire unexpectedly and so that you can rotate as necessary. 
```
Use "like" when comparing things and "such as" when you are providing an example. Use present tense and active per the style guide.

> ** Why they're important:** When using an external system, such as Vault, to manage the CA, you should monitor renewals to ensure that certificates do not expire unexpectedly and so that you can rotate as necessary. 

Reviewer 2: Don't you always need to monitor `consul.mesh.active-root-ca.expiry` no matter what, since Consul will never rotate that on your behalf (whether or not you're using the Vault provider)?

It also seems like you'd want to monitor the signing CA no matter what, just in case the automatic renewal fails for whatever reason, which is more likely if using an external provider (such as Vault).

Author: I was under the impression, mistakenly it seems, that the built-in CA handled rotation for you. IE. that you only needed to worry about rotation when using an external provider.

I'll rework this with that in light.

> ** What to look for:** You usually want to rotate your CA with time to spare, so set a watch based on when you would like it to be rotated.

Reviewer: ```suggestion
** What to look for:** Rotate your CA prior to expiration by setting a watch based on a rotation schedule.
```
Is that more concrete?

> ** What to look for:** Rotate your CA prior to expiration by setting a watch based on a rotation schedule.

Reviewer 2: The action here depends on whether it's a root or an intermediate CA (similar to the comment above). It might be best to link out to a section in the CA docs that describes this in more detail, rather than trying to include it on the telemetry page (which would just continue to make this page more massive than it already is).

Author: The best link I've found is the mesh ca overview...

https://developer.hashicorp.com/consul/docs/connect/ca

Though it doesn't really have a good target to link to which covers these ideas directly. The current version (just pushed) doesn't have the link as I tried to do without due to that.

**What to look for:** For a healthy cluster, you're looking for a `lastContact` lower than 200ms, `leader` > 0 and `candidate` == 0. Deviations from this might indicate flapping leadership.

### Certificate Authority Expiration

| Metric Name                            | Description                                                             | Unit    | Type  |
| :------------------------- | :---------------------------------------------------------------------------------- | :------ | :---- |
| `consul.mesh.active-root-ca.expiry`    | The number of seconds until the root CA expires, updated every hour.    | seconds | gauge |
| `consul.mesh.active-signing-ca.expiry` | The number of seconds until the signing CA expires, updated every hour. | seconds | gauge |

** Why they're important:** Consul Mesh requires a CA to sign all certificates
used to connect the mesh and the mesh network ceases to work if they expire and
become invalid. The Root is particularly important to monitor as Consul does
not automatically rotate it.

** What to look for:** The Root CA should be monitored for an approaching

> expiration, to indicate it is time to manually rotate the CA or to verify your
> automated system is working. The signing (intermediate) certificate should be

Reviewer 2: It seems possible "your automated system" could be misinterpreted as "Consul". Here's my attempt to disambiguate.

```suggestion
expiration, to indicate it is time for you to rotate the "root" CA either manually or with external automation.
The signing (intermediate) certificate should be
```

> rotated automatically by Consul but it is good to monitor it in case of issues.

Reviewer 2: Optional clarification to consider
```suggestion
rotated automatically by Consul, but should be monitored in case of rotation isn't working;
in this scenario, check the server agent logs for messages related to the CA system.
```
